cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedStreamSizeScheduleStore.java

```diff
@@ -311,15 +334,33 @@ public void apply() throws Exception {
           if (txMethod != null) {
             txMethod.execute();
           }
-          byte[] rowKey = Bytes.toBytes(String.format("%s:%s", KEY_PREFIX,
-                                                      AbstractSchedulerService.scheduleIdFor(programId, programType,
-                                                                                             scheduleName)));
+          byte[] rowKey = Bytes.toBytes(getRowKey(programId, programType, scheduleName));
           table.put(rowKey, columns, values);
           LOG.debug("Updated schedule {} with columns {}, values {}", scheduleName, columns, values);
         }
       });
   }
 
+  @Nullable
+  String removeAppVersion(String scheduleId) {
+    String[] parts = scheduleId.split(":");
+    // New Row key for the trigger should be of the form -
+    // streamSizeSchedule:namespace:application:version:type:program:schedule
+    if (parts.length != 7) {
+      return null;
+    }
+
+    // If the version id is not the same as Default version, then return null since we don't want to delete the version
+    // less row key of that schedule
+    if (!parts[3].equals(ApplicationId.DEFAULT_VERSION)) {
+      return null;
+    }
+
+    List<String> stringParts = new ArrayList<>(Arrays.asList(parts));
```

Comment:
since the usages of this method is converting the string to bytes, might be better for this method to return `byte[]` then you can just use `Bytes.add` api to concatenate interested fields.